azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -897,28 +897,33 @@ class ReconnectState(object):
     """
     Class which holds reconenct states as class variables.  Created to make code that reads like an enum without using an enum.
 
-    NEVER_CONNECTED: Ttransport has never been conencted.  This state is necessary because some errors might be fatal or transient,
-    depending on wether the transport has been connceted.  For example, a failed conenction is a transient error if we've connected
-    before, but it's fatal if we've never conencted.
+    WAITING_TO_RECONNECT: This stage is in a waiting period before reconnecting.  This state implies
+    that the user wants the pipeline to be connected.  ie. After a successful connection, the
+    state will change to LOGICALLY_CONNECED
 
-    WAITING_TO_RECONNECT: This stage is in a waiting period before reconnecting.
+    LOGICALLY_CONNECTED: The client wants the pipeline to be connected.  This state is independent
+    of the actual connection state since the pipeline could be logically connected but
+    physically disconnected.
 
-    CONNECTED_OR_DISCONNECTED: The transport is either connected or disconencted.  This stage doesn't really care which one, so
-    it doesn't keep track.
+    LOGICALLY_DISCONNECTED: The client does not want the pipeline to be connected or the pipeline had
+    a fatal error and was forced to disconnect.  If the state is LOGICALLY_DISCONNECTED, then the pipeline
+    should be physically disconnected since there is no reason to leave the pipeline connected in this state.
     """
 
-    NEVER_CONNECTED = "NEVER_CONNECTED"
     WAITING_TO_RECONNECT = "WAITING_TO_RECONNECT"
-    CONNECTED_OR_DISCONNECTED = "CONNECTED_OR_DISCONNECTED"
+    LOGICALLY_CONNECTED = "LOGICALLY_CONNECTED"
+    LOGICALLY_DISCONNECTED = "LOGICALLY_DISCONNECTED"
 
 
 class ReconnectStage(PipelineStage):
     def __init__(self):
         super(ReconnectStage, self).__init__()
         self.reconnect_timer = None
-        self.state = ReconnectState.NEVER_CONNECTED
+        self.state = ReconnectState.LOGICALLY_DISCONNECTED
+        self.never_connected = True
         # connect delay is hardcoded for now.  Later, this comes from a retry policy
-        self.reconnect_delay = 10
+        self.delayed_reconnect_delay = 10
```

Comment:
>delayed_reconnect_delay [](start = 13, length = 23)

could you put some help docs over the two types of delay ? when do we choose one over the other ?

Reply:
+1

---
In reply to: [437135351](https://github.com/Azure/azure-iot-sdk-python/pull/576#discussion_r437135351) [](ancestors = 437135351)

Reply:
I went back to one delay variable and hardcoded the other.  There's no reason to ever adjust the immediate delay and it's only used in a single case, so it seems less confusing if we only have a single delay variable.


---
In reply to: [439658750](https://github.com/Azure/azure-iot-sdk-python/pull/576#discussion_r439658750) [](ancestors = 439658750,437135351)